gen-app-builder/data-store-status-checker/examples/data_store_checker.ipynb

```diff
@@ -0,0 +1,433 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #25.            # implement a rate_limit to prevent quota exhaustion
</pre><p>Maybe you could simplify this loop so that it doesn't need to use the same code multiple times?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/139/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
I refactored the code to package the request/response call into a separate method to be easily reused.  
We'll still need to call `list_documents` 1 time outside of the loop to check for `next_page_token` before attempting to enter the loop.  

It's really the only way to implement a `rate_limit` method that I could find.  
Otherwise, if we only call `list_documents` once and try to extract the contents of the Pager Iterable into a List, we'll hit a quota exhaustion limit as it iters through the Pager to retrieve all the docs.